### PR TITLE
EZP-30921: Dropped deprecated eZc database abstraction layer

### PR DESCRIPTION
**Pull request by alongosz**
| Question                                  | Answer
| ---------------------------------------- | ------------------
| **JIRA issue**                          | [EZP-30921](https://jira.ez.no/browse/EZP-30921)
| **Type**                                   | improvement
| **Target eZ Platform version** | `v3.0`
| **BC breaks**                          | yes
| **Tests pass**                          | yes
| **Doc needed**                       | yes

This PR contains cumulative miscellaneous changes needed in order to completely drop eZc Database Handler (a.k.a Zeta Components Database abstraction layer) from eZ Platform Kernel.

It is blocked by #28 and thus contains it.
The current range for review is: https://github.com/ezsystems/ezplatform-kernel/pull/11/files/c787764aaf9d67af214782994f7a0beb09735e61.

#### Checklist:
- [x] All parts of [EZP-30921](https://jira.ez.no/browse/EZP-30921) are completed.
- [x] Tests pass.
- [x] Code follows Coding Standards (use `$ composer fix-cs`).
- [x] PR is ready for a review.

### CODE REVIEW

File: eZ/Publish/SPI/FieldType/FieldStorage.php [hunk 1]
```diff
@@ -16,16 +14,7 @@
...
- * currently used storage engine.
- * The array should at least define 2 keys :
- *   - identifier (connection identifier)
- *   - connection (the connection handler)
- * For example, using Legacy storage engine, $context will be:
- *   - identifier = 'LegacyStorage'
- *   - connection = {@link \eZ\Publish\Core\Persistence\Doctrine\ConnectionHandler} object handler (for DB connection),
- *                  to be used accordingly to
- *                  {@link http://incubator.apache.org/zetacomponents/documentation/trunk/Database/tutorial.html ezcDatabase} usage
+ * $context array passed to most methods is deprecated and will be dropped in the next major version.
```
Comment (Nattfarinn) [inline eZ/Publish/SPI/FieldType/FieldStorage.php:17]: Isn't now the best time to drop it, and push deprecation to current LTS?

File: doc/bc/changes-1.0.md [hunk 2]
```diff
@@ -262,6 +262,13 @@ Changes affecting version compatibility with deprecated ezpublish-kernel version
...
 * Legacy (SQL) Search Engine will no longer treat deprecated `%` as a search wildcard. Use `*` instead.
 
+* Field Type External Storage Handlers `$context` array no longer has the "connection" key. Rely on
+  injected Connection instead.
+
+* The deprecated Zeta Components (eZc) Database handler has been dropped. All classes and interfaces
+  from the namespaces `eZ\Publish\Core\Persistence\Database` and `eZ\Publish\Core\Persistence\Doctrine`
+  were removed.
+
 ## Deprecated features
```
Comment (Nattfarinn) [inline doc/bc/changes-1.0.md:273]: There's array context of FieldStorage deprecated. Not a feature per se, but may be worth noting here.

File: eZ/Publish/SPI/FieldType/FieldStorage.php (hunk 1 above)
Comment (alongosz) [reply eZ/Publish/SPI/FieldType/FieldStorage.php:17]: Unfortunately would require SPI breaking changes to ~36 methods in our OSS code, not to mention  1st party packages providing Field Types. Planned to do it as a follow-up, but it's too late now :( Gonna add a BC note though.

File: doc/bc/changes-1.0.md (hunk 2 above)
Comment (alongosz) [reply doc/bc/changes-1.0.md:273]: Added via f3f4eb2

**Pull request merged**